test/integration/overload_integration_test.cc

```diff
@@ -118,4 +133,34 @@ TEST_P(OverloadIntegrationTest, DisableKeepaliveWhenOverloaded) {
   EXPECT_EQ(nullptr, response->headers().Connection());
 }
 
+TEST_P(OverloadIntegrationTest, StopAcceptingConnectionsWhenOverloaded) {
+  initialize();
+  fake_upstreams_[0]->set_allow_unexpected_disconnects(true);
+
+  // Put envoy in overloaded state and check that it doesn't accept the new client connection.
+  updateResource(0.95);
+  test_server_->waitForGaugeEq("overload.envoy.overload_actions.stop_accepting_connections.active",
+                               1);
+  codec_client_ = makeHttpConnection(makeClientConnection((lookupPort("http"))));
+  Http::TestHeaderMapImpl request_headers{
+      {":method", "GET"}, {":path", "/test/long/url"}, {":scheme", "http"}, {":authority", "host"}};
+  auto response = codec_client_->makeRequestWithBody(request_headers, 10);
+  EXPECT_FALSE(response->waitForEndStreamWithTimeout(std::chrono::milliseconds(2000)));
```

Comment:
Can we have a shorter wait time here in the interests of our tests being fast?

Though now that I think on it, if you disabled your overload code wouldn't this test still pass? Nothing is set up to end the stream in any case, so we wouldn't have an end stream unless we waited 5s for the network idle timeout to kick in.

I basically hate sleep() tests but basically what we want to do here is simply wait for some period and then verify that no new incoming connections have been accepted, right?

Reply:
Good catch - thanks! Yes, this part of the test just wants to verify that no new incoming connections are accepted while envoy is overloaded. I changed it to wait (and expect a timeout) for an upstream http connection to be established.